project/backend/src/main/java/com/google/sps/data/RestaurantScorer.java

```diff
@@ -31,16 +34,38 @@ public static double score(Restaurant restaurant, JSONObject preferences) throws
     // MID_RATING.
     boolean hasRating = restaurantRating != -1;
     if (hasRating) {
-      score += calculateRatingScore(restaurantRating, restaurant.getNumRatings());
+      score += calculateRatingScore(restaurantRating, restaurant.getNumRatings(), statistics);
     }
     double percentMatch = score / maxPoints;
     return percentMatch;
   }
 
+  /**
+   * Creates and returns a DescriptiveStatistics object containing all the values for number of
+   * ratings for each restaurant.
+   */
+  public static DescriptiveStatistics createDescriptiveStatistics(JSONArray restaurantList) {
+    DescriptiveStatistics statistics = new DescriptiveStatistics();
+    for (int i = 0; i < restaurantList.length(); i++) {
+      try {
+        // int numRatings = restaurantList.getJSONObject(i).has("user_ratings_total") ?
+        // restaurantList.getJSONObject(i).getInt("user_ratings_total") : 0;
+        // statistics.addValue(numRatings);
+        statistics.addValue(restaurantList.getJSONObject(i).getInt("user_ratings_total"));
+      } catch (JSONException e) {
+        // LOGGER.log("Error parsing JSON: " + e.getMessage());
```

Comment:
remove commented out lines?

Reply:
ohh I meant to leave a comment about this-- I was just wondering which would be a better way to code this. The commented out lines are an alternative way (and then I wouldn't keep lines 54 and 57) to do the same thing. Does one look better than the other?

Reply:
you'd still want to keep addValue(0) in the catch block right? or does the commented out way mean the exception wont need to be handled?

Reply:
51-52 would cover that because it checks to see if user_ratings_total is a field and if it is not, then sets numRatings to 0. So the catch block would only execute in the case of an unexpected error.

Reply:
I see - i think we need addValue(0) in either case then : when user rating is not a field and when there';s an unexpected error so:
```
try {
  JSONObject restaurantObj =  restaurantList.getJSONObject(i);
  int numRatings = restaurantObj.has("user_ratings_total") ?restaurantObj.getInt("user_ratings_total") : 0;
  statistics.addValue(numRatings);
} catch (JSONException e) {
    statistics.addValue(0);
}
```